graphbolt/src/cuda/neighbor_sampler.cu

```diff
@@ -519,9 +520,7 @@ c10::intrusive_ptr<sampling::FusedSampledSubgraph> SampleNeighbors(
           }
         }
 
-        // TODO @mfbalin: remove true from here once fetching indices later is
-        // setup.
-        if (true || layer || utils::is_on_gpu(indices)) {
+        if (!returning_indices_is_optional || utils::is_on_gpu(indices)) {
```

Comment:
Why not we move this 2 check to python, such that we only need to pass a deterministic option down?

Reply:
I think it is already deterministic. I coded it this way to ensure puregpu is not affected.

Reply:
Yes, I know the code pass is deterministic. The problem comes from the parameter name, it mis-led the user. After several months when we look back to the code, we will get trapped.

It is ok to leave this as it, and finish the primary work first.

Reply:
When I try to understand a deterministic statement with the word `maybe` in it, I will be confused.